### PR TITLE
Shutdown Host Action

### PR DESCRIPTION
**Pull request by aalku**
### Description

This adds a new "Shutdown Host" menu entry to launch an action to request the host (OctoPrint) to shutdown before manually power off the 3D printer.

When you use OctoPi in a Raspberry Pi that is powered with the 3D Printer PSU then if you just cut the current you will eventually corrupt the Rapsberry Pi SD card so you should always shutdown OctoPi before powering down. But that's not always convenient. Maybe you just powered on the 3D Printer to change filament and then you want to power it off without starting the computer to tell OctoPi to shutdown. So this feature allows you to tell the host to shutdown from the very Marlin menu.

### Requirements

In order to use it the host has to support it. In the case of OctoPi you can enable it with the [Action Commands plugin](https://github.com/benlye/OctoPrint-ActionCommandsPlugin) configured like this:
image

That is **Action** `shutdown` **Type** `SYSTEM` and **Command** `sudo shutdown -h now`

You need an LCD with standard Marlin UI menu as I only added it to it.

### Benefits

This feature improves the 3D Printer usability when you have a embeded host sharing the PSU with the Printer. 

### Configurations

You need to uncomment `#define HOST_ACTION_COMMANDS` and `#define HOST_SHUTDOWN_MENU_ITEM` lines in `Marlin\Configuration_adv.h` file. If you use your own file just uncomment the first and add the second.

### Related Issues

Nothing here

### CODE REVIEW

**Comment by aalku** [comment]
I don't know if I'm making the pull request right as I guess the tests are run with default or predefined configuration files so I think automatic tests are not testing that it compiles with HOST_ACTION_COMMANDS and HOST_SHUTDOWN enabled. It compiles and works for me 😅 but there is no independent evidence.

File: Marlin/src/lcd/language/language_en.h [hunk 1]
```diff
@@ -745,6 +745,10 @@ namespace Language_en {
   LSTR MSG_SD_CARD                        = _UxGT("SD Card");
   LSTR MSG_USB_DISK                       = _UxGT("USB Disk");
 
+  #if ENABLED(HOST_SHUTDOWN)
+    PROGMEM Language_Str SHUTDOWN_HOST    = _UxGT("Shutdown Host");
+  #endif
```
Comment (InsanityAutomation) [inline Marlin/src/lcd/language/language_en.h]: as a command, this should be static text, not translated. The command will not be parsed by the host side as a translated string.

Comment (aalku) [reply Marlin/src/lcd/language/language_en.h]: That's not the action command, that's the UI menu item to give the order.

Comment (aalku) [reply Marlin/src/lcd/language/language_en.h]: This one is the action command.

https://github.com/MarlinFirmware/Marlin/blob/50c2c9e0ac1ce5fefbdadc89050d3348b1839eb5/Marlin/src/inc/Conditionals_adv.h#L705

Comment (InsanityAutomation) [reply Marlin/src/lcd/language/language_en.h]: Then I have not had enough coffee yet this morning lol

**Comment by thinkyhead** [comment]
> I guess the tests are run with default or predefined configuration files

The files in `buildroot/tests` set the options used for build tests, so you can always add the new options to one of those test files to get it building under CI.

**Comment by aalku** [comment]
I updated the PR description according to your refinement changes.
It works fine as it is but for one thing. When you confirm the shutdown it sends the action but the submenu to confirm remains open. You have to cancel to exit but I think it should exit that screen after executing the shutdown.

I don't know how to do it.

**Comment by thinkyhead** [comment]
> I don't know how to do it.

I poked around the code and found several examples. I also noticed that I had set up the select screen to take a `nullptr` for a return to previous screen, so I went ahead and tweaked several other select screen calls.

**Comment by aalku** [comment]
It works perfectly! Thank you!

Anything else needed in order to merge?

**Comment by aalku** [comment]
Thank you!

**Pull request merged**